script/releases/v1.0.0-slashing/2-queueUpgradeAndUnpause.s.sol

```diff
@@ -0,0 +1,148 @@
+            /// core/
+            .append({
+                to: Env.proxyAdmin(),
+                data: Encode.proxyAdmin.upgrade({
+                    proxy: address(Env.proxy.avsDirectory()),
+                    impl: address(Env.impl.avsDirectory())
+                })
+            })
+            .append({
+                to: Env.proxyAdmin(),
+                data: Encode.proxyAdmin.upgrade({
+                    proxy: address(Env.proxy.delegationManager()),
+                    impl: address(Env.impl.delegationManager())
+                })
+            })
+            .append({
+                to: Env.proxyAdmin(),
+                data: Encode.proxyAdmin.upgrade({
+                    proxy: address(Env.proxy.rewardsCoordinator()),
+                    impl: address(Env.impl.rewardsCoordinator())
+                })
+            })
+            .append({
+                to: Env.proxyAdmin(),
+                data: Encode.proxyAdmin.upgrade({
+                    proxy: address(Env.proxy.strategyManager()),
+                    impl: address(Env.impl.strategyManager())
+                })
+            })
+            /// pods/
+            .append({
+                to: address(Env.proxy.eigenPod()),
+                data: Encode.upgradeableBeacon.upgradeTo({
+                    newImpl: address(Env.impl.eigenPod())
+                })
+            })
+            .append({
+                to: Env.proxyAdmin(),
+                data: Encode.proxyAdmin.upgrade({
+                    proxy: address(Env.proxy.eigenPodManager()),
+                    impl: address(Env.impl.eigenPodManager())
+                })
+            })
+            /// strategies/
+            .append({
+                to: Env.proxyAdmin(),
+                data: Encode.proxyAdmin.upgrade({
+                    proxy: address(Env.proxy.eigenStrategy()),
+                    impl: address(Env.impl.eigenStrategy())
+                })
+            })
+            .append({
+                to: address(Env.proxy.strategyBase()),
+                data: Encode.upgradeableBeacon.upgradeTo({
+                    newImpl: address(Env.impl.strategyBase())
+                })
+            })
+            .append({
+                to: Env.proxyAdmin(),
+                data: Encode.proxyAdmin.upgrade({
+                    proxy: address(Env.proxy.strategyFactory()),
+                    impl: address(Env.impl.strategyFactory())
+                })
+            });
+
+        /// Add call to upgrade each pre-longtail strategy instance
+        uint count = Env.instance.strategyBaseTVLLimits_Count();
+        for (uint i = 0; i < count; i++) {
+            address proxyInstance = address(Env.instance.strategyBaseTVLLimits(i));
+
+            executorCalls.append({
+                to: Env.proxyAdmin(),
+                data: Encode.proxyAdmin.upgrade({
+                    proxy: proxyInstance,
+                    impl: address(Env.impl.strategyFactory())
```

Comment:
Wrong upgrade implementation

Reply:
Should be new `StrategyBase`

Reply:
This is why I find the `_verify...` functions in ExistingDeploymentParser to be useful as part of sanity check tests. We should port those over.

Reply:
Can you link? What do these do?

Reply:
@ypatil12 new `StrategyBase`? Or new `StrategyBaseTVLLimits`?

I was gonna do the latter but did we discuss doing the former? (theoretically we aren't using and don't plan on using the "tvl limits" part of that contract, but i havent checked for storage compatibility)